contracts/protocol/v1/proxies/P1LiquidatorProxy.sol

```diff
@@ -0,0 +1,285 @@
+ * directed to the insurance fund.
+ */
+contract P1LiquidatorProxy is
+    Ownable
+{
+    using BaseMath for uint256;
+    using SignedMath for SignedMath.Int;
+    using P1BalanceMath for P1Types.Balance;
+
+    // ============ Events ============
+
+    event LogLiquidatorProxyUsed(
+        address indexed liquidator,
+        address indexed liquidatee,
+        bool isBuy,
+        uint256 liquidationAmount,
+        uint256 feeAmount
+    );
+
+    event LogInsuranceFundSet(
+        address insuranceFund
+    );
+
+    // ============ Immutable Storage ============
+
+    // Address of the perpetual contract.
+    address public _PERPETUAL_V1_;
+
+    // Address of the P1Liquidation contract.
+    address public _LIQUIDATION_;
+
+    // Percentage of fees that go to the insurance fund as a fixed-point number with 18 decimals.
+    uint256 public _FEE_PERCENTAGE_;
+
+    // ============ Mutable Storage ============
+
+    // Address of the insurance fund.
+    address public _INSURANCE_FUND_;
+
+    // ============ Constructor ============
+
+    constructor (
+        address perpetualV1,
+        address liquidator,
+        address insuranceFund,
+        uint256 feePercentage
+    )
+        public
+    {
+        _PERPETUAL_V1_ = perpetualV1;
+        _LIQUIDATION_ = liquidator;
+        _INSURANCE_FUND_ = insuranceFund;
+        _FEE_PERCENTAGE_ = feePercentage;
+
+        emit LogInsuranceFundSet(insuranceFund);
+    }
+
+    // ============ External Functions ============
+
+    /**
+     * @notice Sets the maximum allowance on the perpetual contract. Must be called once.
+     * @dev Cannot be run in the constructor due to technical restrictions in Solidity.
+     */
+    function setAllowance()
+        external
+    {
+        address perpetual = _PERPETUAL_V1_;
+        address tokenContract = I_PerpetualV1(perpetual).getTokenContract();
+        SafeERC20.safeApprove(
+            IERC20(tokenContract),
+            perpetual,
+            uint256(-1)
+        );
+    }
+
+    /**
+     * @notice Allows an account below the minimum collateralization to be liquidated by another
+     *  account. This allows the account to be partially or fully subsumed by the liquidator.
+     * @dev Emits the LogLiquidatorProxyUsed event.
+     *
+     * @param  liquidatee   The account to liquidate.
+     * @param  isBuy        True if the liquidatee has a long position, false otherwise.
+     * @param  maxPosition  Maximum position size that the liquidator will take post-liquidation.
+     * @return              The change in position.
+     */
+    function liquidate(
+        address liquidatee,
+        bool isBuy,
+        SignedMath.Int calldata maxPosition
+    )
+        external
+        returns(uint256)
+    {
+        I_PerpetualV1 perpetual = I_PerpetualV1(_PERPETUAL_V1_);
+
+        // Get the balances of the sender.
+        perpetual.deposit(msg.sender, 0);
```

Comment:
Could you explain why you're depositing 0 here?

Reply:
Will this result in a deposit log being emitted?

Reply:
To first settle the balances of the account. It will, yes

Reply:
It might, but it's not really useful unless you're going to settle the account within that transaction anyway (as the funding rate may change otherwise). There isn't an ton of wasted gas here and is much less error-prone